crypto/rand/rand_unix.c

```diff
@@ -15,6 +15,7 @@
 #include "rand_lcl.h"
 #include "internal/rand_int.h"
 #include <stdio.h>
+#include "internal/dso.h"
```

Comment:
There is mismatch. If dso.h is internal, then why are symbols exported? Obviously a side note...

Reply:
??? DSO_bind_func to zlib is made from libcrypto, as far as I can see. And gost is an engine, right? In such case you tell libcrypto to load engine, so it's again libcrypto that calls DSO_\*... Still side note :-)

Reply:
I think we do things like that to be able to (potentially) use it in libssl. It obviously isn't used currently, so the symbol could be removed if we really wanted to.

Reply:
Note that I'm only saying that there is mismatch. As for exporting vs. not exporting, it can be useful even to application. Consider the last controversy about locale dependency. In case we find ourselves deciding to call iconv from apps/\*, wouldn't it be appropriate to detect iconv with DSO_global_lookup ;-)